pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -46,8 +46,12 @@ func (p *Pipeline) SupportedVerbs() []admissionregistrationv1.OperationType {
 // that any references resources exist, that is done at run time.
 func (p *Pipeline) Validate(ctx context.Context) *apis.FieldError {
 	errs := validate.ObjectMetadata(p.GetObjectMeta()).ViaField("metadata")
-	ctx = config.SkipValidationDueToPropagatedParametersAndWorkspaces(ctx, false)
-	return errs.Also(p.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec"))
+	errs = errs.Also(p.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec"))
+	// When a Pipeline is created directly, instead of declared inline in a PipelineRun,
+	// we do not support propagated parameters and workspaces.
+	// Validate that all params and workspaces it uses are declared.
+	errs = errs.Also(p.Spec.validatePipelineParameterUsage().ViaField("spec"))
+	return errs.Also(p.Spec.validatePipelineWorkspacesUsage().ViaField("spec"))
```

Comment:
One thing that I realized while implementing a fix for https://github.com/tektoncd/pipeline/issues/6670 is that here we also need to extract `ValidatePipelineTasks` out of the `Validate` function. This is because propagation dives into the pipeline tasks as well.

Reply:
I wonder if that belongs as a part of this PR or its own separate one.

Reply:
The issue here is that while we didn't initially wrap `ValidatePipelineTask` into this `if condition`, the same `if condition`  was invoked at a later point (via task's validation). So functionally, it was still invoking this condition but that was hidden away from here.

I think that's fine. What I can do is implement this as a part of my PR that is fixing `accidental propagation in referenced pipeline` after this has been merged.

Reply:
Chatted offline with Chitrang-- he is going to check test coverage for propagation to pipeline tasks and add it if needed to make sure this PR truly has no functional changes.

Reply:
Opened a PR to address this: https://github.com/tektoncd/pipeline/pull/6710